### PR TITLE
applied fade in effect on spawn

### PR DESCRIPTION
**Pull request by EdwardRegasDuncan**

### CODE REVIEW

File: Gameplay/entities/enemies/EnemyController.cpp [hunk 1]
```diff
@@ -786,7 +794,7 @@ void Hachiko::Scripting::EnemyController::Spawn()
 		if (_enemy_body)
 		{
 			_enemy_body->SetActive(true);
-			_enemy_body->ChangeEmissiveColor(float4(0.3f, 0.5f, 1.0f, 1.0f), _spawning_time, true);
+			//_enemy_body->ChangeEmissiveColor(float4(0.3f, 0.5f, 1.0f, 1.0f), _spawning_time, true);
```
Comment (baransrc) [inline Gameplay/entities/enemies/EnemyController.cpp:797]: Let's delete this comment.

File: Gameplay/entities/enemies/EnemyController.h [hunk 2]
```diff
@@ -178,6 +178,8 @@ namespace Hachiko
             float3 _knockback_pos = float3::zero;
             float _attack_current_delay = 0.f;
             SERIALIZE_FIELD(float, _current_spawning_time);
+            float spawn_progress = 0.0f;
+            const float spawn_rate = 1 / math::Sqrt(_spawning_time);
```
Comment (baransrc) [inline Gameplay/entities/enemies/EnemyController.h:182]: Why division by Sqrt?

File: Gameplay/entities/enemies/EnemyController.cpp [hunk 3]
```diff
@@ -118,6 +118,14 @@ void Hachiko::Scripting::EnemyController::OnUpdate()
 
 	if (_current_spawning_time > 0.0f || _state == EnemyState::SPAWNING)
 	{
+		spawn_progress += spawn_rate * Time::DeltaTime();
+
+
+		if (spawn_progress <= 1.0f) {
+			_enemy_body->ChangeTintColor(float4(1.0f, 1.0f, 1.0f, spawn_progress), true);
+		}
```
Comment (baransrc) [inline Gameplay/entities/enemies/EnemyController.cpp:126]:
```suggestion
		spawn_progress += spawn_rate * Time::DeltaTime();
		spawn_progress = spawn_progress > 1.0f ? 1.0f : spawn_progress;
		
		_enemy_body->ChangeTintColor(float4(1.0f, 1.0f, 1.0f, spawn_progress), true);
```

File: Gameplay/entities/enemies/EnemyController.cpp (hunk 1 above)
Comment (baransrc) [inline Gameplay/entities/enemies/EnemyController.cpp:797]:
```suggestion
```

**Review by baransrc — CHANGES_REQUESTED** [review]
Thanks for the work!! I requested small changes and a clarification, afterwards we are good to go.

**Review by bernatgb — COMMENTED** [review]
Good job, but it feel a bit weird see the enemies appear, disappear, and then spawning with the fade effect.

Apart from that, sometimes the enemies remain invisible:
bugbug
_I'm in god mode here, but it still happens even if I turn it off._

File: Gameplay/entities/enemies/EnemyController.h [hunk 4]
```diff
@@ -181,6 +181,12 @@ namespace Hachiko
             // Combat info
             float3 _knockback_pos = float3::zero;
             float _attack_current_delay = 0.f;
+            SERIALIZE_FIELD(float, _current_spawning_time);
```
Comment (baransrc) [inline Gameplay/entities/enemies/EnemyController.h:184]: Let's remove this if we are not using it.

File: Gameplay/entities/enemies/EnemyController.h [hunk 5]
```diff
@@ -181,6 +181,12 @@ namespace Hachiko
...
+            float spawn_progress = 0.0f;
+            const float spawn_rate = 1.0f;
+            //float _current_spawning_time = 0.f;
+            bool _has_spawned = false;
```
Comment (baransrc) [inline Gameplay/entities/enemies/EnemyController.h:188]: Let's remove this if we are not using.

**Review by baransrc — CHANGES_REQUESTED** [review]

**Review by baransrc — APPROVED** [review]

**Review by Bernatmago — APPROVED** [review]
Works like a charm thanks!
A PR description saying what change is expected to be noticeable and how to test it, even if very brief, would be greatly appreciated in the future!

**Pull request merged**